### PR TITLE
Upgrade to support dbt core v141

### PR DESCRIPTION
**Pull request by damian3031**
Closes https://github.com/starburstdata/dbt-trino/issues/213

### CODE REVIEW

File: .changes/unreleased/Breaking Changes-20230126-200731.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+kind: Breaking Changes
```
Comment (hovaesco) [inline .changes/unreleased/Breaking Changes-20230126-200731.yaml:1]: Move it to `Features` or `Under the hood`

**Review by hovaesco — COMMENTED** [review]
New tests introduced in 1.4.0 are missing.

File: .changes/unreleased/Breaking Changes-20230126-200731.yaml (hunk 1 above)
Comment (damian3031) [reply .changes/unreleased/Breaking Changes-20230126-200731.yaml]: changed to `Under the Hood`

**Comment by damian3031** [comment]
> New tests introduced in 1.4.0 are missing.

@hovaesco 
All new tests from `dbt-core` 1.4.0 added

File: dbt/adapters/trino/relation.py [hunk 2]
```diff
@@ -13,7 +13,7 @@ class TrinoQuotePolicy(Policy):
 
 @dataclass(frozen=True, eq=False, repr=False)
 class TrinoRelation(BaseRelation):
-    quote_policy: TrinoQuotePolicy = TrinoQuotePolicy()
+    quote_policy: TrinoQuotePolicy = field(default_factory=lambda: TrinoQuotePolicy())
```
Comment (findinpath) [inline dbt/adapters/trino/relation.py:16]: Is this change related to the upgrade?

File: dbt/include/trino/macros/materializations/incremental.sql [hunk 3]
```diff
@@ -86,9 +86,9 @@
     {% endif %}
 
     {#-- Get the incremental_strategy, the macro to use for the strategy, and build the sql --#}
-    {% set incremental_predicates = config.get('incremental_predicates', none) %}
+    {% set incremental_predicates = config.get('predicates', none) or config.get('incremental_predicates', none) %}
```
Comment (findinpath) [inline dbt/include/trino/macros/materializations/incremental.sql:89]: Note that this change is rather characteristic for dbt-core 1.4.1 and not dbt-core 1.4.0

See https://github.com/dbt-labs/dbt-core/commit/e8da84fb9e177d9eee3d7722d3d6906bb283183d

File: README.md [hunk 4]
```diff
@@ -39,36 +39,36 @@ $ pip install dbt-trino
...
+| jwt_token                      | JWT token for authentication                                                                                 | Optional (required if `method` is `jwt`)                                                                | `none` or `abc123`               |
+| client_certificate             | Path to client certificate to be used for certificate based authentication                                   | Optional (required if `method` is `certificate`)                                                        | `/tmp/tls.crt`                   |
+| client_private_key             | Path to client private key to be used for certificate based authentication                                   | Optional (required if `method` is `certificate`)                                                        | `/tmp/tls.key`                   |
+| http_headers                   | HTTP Headers to send alongside requests to Trino, specified as a yaml dictionary of (header, value) pairs.   | Optional                                                                                                | `X-Trino-Client-Info: dbt-trino` |
```
Comment (findinpath) [inline README.md:61]: Pls add a separate commit for the spacing changes in the README.md

File: README.md [hunk 5]
```diff
@@ -39,36 +39,36 @@ $ pip install dbt-trino
...
-| threads                        | How many threads dbt should use                                                                              | Optional (default is `1`)                                                                               | `8`                                      |
-| prepared_statements_enabled    | Enable usage of Trino prepared statements (used in `dbt seed` commands)                                      | Optional (default is `true`)                                                                            | `true` or `false`                        |
-| retries                        | Configure how many times a database operation is retried when connection issues arise                        | Optional (default is `3`)                                                                               | `10`                                     |
+| Option                         | Description                                                                                                  | Required?                                                                                               | Example                          |
```
Comment (findinpath) [inline README.md:42]: btw, you could extract this change to a separate PR which is much easier to grasp and to merge

File: .changes/unreleased/Under the Hood-20230130-092157.yaml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+kind: Under the Hood
+body: Adjustment to dbt-core 1.4.0
```
Comment (findinpath) [inline .changes/unreleased/Under the Hood-20230130-092157.yaml:2]: Why not 1.4.1 ?

File: dbt/adapters/trino/relation.py (hunk 2 above)
Comment (damian3031) [reply dbt/adapters/trino/relation.py:16]: yes, because of [this change](https://github.com/dbt-labs/dbt-core/blob/v1.4.1/core/dbt/adapters/base/relation.py#L33-L36)

File: README.md (hunk 4 above)
Comment (damian3031) [reply README.md:61]: Extracted to separete commit

File: .changes/unreleased/Under the Hood-20230130-092157.yaml (hunk 6 above)
Comment (damian3031) [reply .changes/unreleased/Under the Hood-20230130-092157.yaml]: Actually this PR is about to adjust to 1.4.1, I changed this here

File: dbt/include/trino/macros/materializations/incremental.sql (hunk 3 above)
Comment (damian3031) [reply dbt/include/trino/macros/materializations/incremental.sql:89]: Actually this PR is about to adjust to 1.4.1

File: tests/functional/adapter/materialization/test_incremental_predicates.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+@pytest.mark.delta
+class TestDeltaIncrementalPredicatesMergeTrino(BaseIncrementalPredicates):
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "models": {
+                "+incremental_predicates": ["dbt_internal_dest.id != 2"],
+                "+incremental_strategy": "merge",
+                "+on_table_exists": "drop",
```
Comment (hovaesco) [inline tests/functional/adapter/materialization/test_incremental_predicates.py:41]: `on_table_exists` probably not needed here.

File: dbt/adapters/trino/column.py [hunk 8]
```diff
@@ -3,7 +3,7 @@
 from typing import ClassVar, Dict
 
 from dbt.adapters.base.column import Column
-from dbt.exceptions import RuntimeException
+from dbt.exceptions import DbtRuntimeError
```
[the remark this answers is not in the document]
Comment (damian3031) [reply dbt/adapters/trino/column.py:6]: that is a breaking change in dbt 1.4.0 https://github.com/dbt-labs/dbt-core/issues/6460

File: tests/functional/adapter/materialization/test_incremental_predicates.py (hunk 7 above)
Comment (damian3031) [reply tests/functional/adapter/materialization/test_incremental_predicates.py]: removed

**Review by hovaesco — APPROVED** [review]

File: setup.py [hunk 9]
```diff
@@ -86,7 +86,7 @@ def _get_dbt_core_version():
     },
     install_requires=[
         "dbt-core~={}".format(dbt_version),
-        "trino==0.319.0",
+        "trino==0.321.0",
```
[the remark this answers is not in the document]
Comment (damian3031) [reply setup.py:89]: added

File: dbt/adapters/trino/connections.py [hunk 10]
```diff
@@ -429,6 +435,7 @@ def open(cls, connection):
             isolation_level=IsolationLevel.AUTOCOMMIT,
             source="dbt-trino",
             verify=credentials.cert,
+            timezone=credentials.timezone,
```
[the remark this answers is not in the document]
Comment (damian3031) [reply dbt/adapters/trino/connections.py:438]: added

**Review by hovaesco — APPROVED** [review]

File: .changes/unreleased/Features-20230130-160859.yaml [hunk 11]
```diff
@@ -0,0 +1,7 @@
+kind: Features
+body: Added new connection property `timezone`
```
Comment (findinpath) [inline .changes/unreleased/Features-20230130-160859.yaml:2]: @damian3031  could you please document `timezone` in `README.md` ?

Comment (findinpath) [reply .changes/unreleased/Features-20230130-160859.yaml:2]: Related PR https://github.com/trinodb/trino-python-client/pull/252
